File: 2024/C_Replacing_Integer.go
// created: 2024/5/28 23:28:51 author:  https://github.com/funcdfs
// problem: https://atcoder.jp/contests/abc161/tasks/abc161_c

package main

import (
	"bufio"
	"fmt"
	"os"
)

func solve(_in *bufio.Reader, _out *bufio.Writer) {
	var n, k uint64
	fmt.Fscan(_in, &n, &k)
	if n > k {
		x := n / k
		if n%k > 0 {
			x += 1
		}
		fmt.Fprintln(_out, x*k-n)
	} else {
		k = k - (k-1)*n
		x := n / k
		if n%k > 0 {
			x += 1
		}
		fmt.Fprintln(_out, x*k-n)
	}

}

func main() {
	_in := bufio.NewReader(os.Stdin)
	_out := bufio.NewWriter(os.Stdout)
	defer _out.Flush()

	solve(_in, _out)
}

File: 2024/C_Replacing_Integer.py
# problem: https://atcoder.jp/contests/abc161/tasks/abc161_c
# created: https://github.com/funcdfs 2024/5/28 23:41:41 

import sys


# region solve
# ------------------------------------------------------------
def solve() -> None:
    n, k = input_multi_int()
    if n >= k:
        x = n // k
        if n % k:
            x += 1
        print(x * k - n)
    else:
        print(n)

# ------------------------------------------------------------
# endregion solve


# region main
input_int = lambda: int(input_str())  # noqa: E731
input_str = lambda: sys.stdin.readline().rstrip()  # noqa: E731
input_multi_int = lambda: map(int, input_str().split())  # noqa: E731
input_list = lambda: list(map(int, input_str().split()))  # noqa: E731
dbg = lambda *args: sys.stderr.write(" ".join(map(str, args)) + "\n")  # noqa: E731

solve()
# endregion main
